architecture/platforms/versal_acap.rst

```diff
@@ -0,0 +1,125 @@
+.. _versal_acap:
+
+#############################################
+VERSAL Adaptive Compute Acceleration Platform
+#############################################
+
+
+.. _overview :
+
+Overview
+********
+
+The OP-TEE support for the AMD/Xilinx Versal Adaptive Compute Acceleration Platform (`Versal ACAP`_) delegates most of its functionality to the `PLM firmware`_ executing in the MicroBlaze processor.
+In the case of cryptographic operations, those operations or ciphers or keys not supported by the hardware will be routed to the `Libtomcrypt`_ software based implementation.
+
+The services offered by PLM firmware are decided at build time: it is therefore important that the PLM firmware build configuration enables the services that OP-TEE will require.
+
+As an example, if OP-TEE is configured to generate a hardware unique key, it  will need access to the PLM Physical Unclonable Function and NVM services.
+
+.. note::
+   Communication between OP-TEE and the PLM uses the IPI mailbox controller being the IPI used selectable via ``CFG_VERSAL_MBOX_IPI_ID``.
+
+As described in freely available documentation for the Versal ACAP `boot-flow`_, the BootROM handles loading the PLM, while the PLM will handle loading the rest of the images including OP-TEE.
+
+A reference BIF file supporting an OP-TEE instance capable of loading the FPGA pdi can be seen below. In this example OP-TEE should be configured with ``CFG_DT=y`` and ``CFG_DT_ADDR=0x00001000``.
+If enabled the platform expects the FPGA bitstream at 0x40000000; the location is configurable using ``CFG_VERSAL_FPGA_DDR_ADDR``.
```

Comment:
This is confusing mentioning the FPGA without first listing all of the Versal specific features in this OP-TEE port.

Reply:
Also link to associated Xilinx libary used. xilfpga.

Reply:
what is confusing about mentioning how to program the FPGA?

Reply:
The main confusion is going into details about programming the FPGA without first finishing the rest of the details of the architecture.

Reply:
That makes no sense: both are orthogonal. Besides the Versal architecture - and the FPGA - should be known to the OP-TEE users. the OP-TEE documentation is not the place to learn about Versal ACAP.